Update 3DS wheel

nrgsille76 opened the pull request:
Updated io_scene_3ds wheel:

Import 3DS:
- Improved texture alpha settings
- Additional texture maps supported

Export 3DS:
- Added new function `save_3ds` matching the `load_3ds` of the importer
- Additional texture maps supported

nrgsille76 commented:
> Thank you for the PR. Do note, that in order to be able to merge this, the [exact same binary wheel must exist on pypi](https://docs.blender.org/manual/en/latest/advanced/extensions/python_wheels.html#extensions-python-wheels) - the docs are not clear on this, but the [moderation requires it](https://extensions.blender.org/approval-queue/open-stage-blender-dmx/#activity-1547).

@vanous  Have done that, it is now on version 2.7.6, I also made new commits for that in this PR.
Here is the [link](https://pypi.org/manage/project/io-scene-3ds/release/2.7.6/) to the binary wheel on pypi

vanous commented:
Thank you. 

I can see that you yourself are using io_scene_3ds-2.7.8-py3-none-any.whl, but this PR is proposing 2.7.6, what is the reason for that?

I am also a bit lost in finding what is the canonical source for io_scene_3ds, out of the following repos which all have various updates: https://github.com/nrgsille76/io_scene_3ds/, https://github.com/nrgsille76/blender-io-addons/, https://projects.blender.org/extensions/io_scene_3ds/.

As for exporting MVR files, [3DS has been deprecated](https://gdtf.eu/mvr/mvr-spec/file-format-definition/#supported-3d-file-formats
).

nrgsille76 commented:
> Thank you.
> 
> I can see that you yourself are using io_scene_3ds-2.7.8-py3-none-any.whl, but this PR is proposing 2.7.6, what is the reason for that?
> 
> I am also a bit lost in finding what is the canonical source for io_scene_3ds, out of the following repos which all have various updates: https://github.com/nrgsille76/io_scene_3ds/, https://github.com/nrgsille76/blender-io-addons/, https://projects.blender.org/extensions/io_scene_3ds/.
> 
> As for exporting MVR files, [3DS has been deprecated](https://gdtf.eu/mvr/mvr-spec/file-format-definition/#supported-3d-file-formats).

I updated the wheel to version 2.8, it was nessecary to check if the collections have hidden viewport for exporting invisble objects.
I also changed it in the PR, but because you seem to do not need the exporter at all it makes no difference.

The canonical source of the wheel is [here](https://github.com/nrgsille76/blender-io-addons/tree/main/io_scene_3ds), the pypi version is slightly different as the extension version, thats why there are more repositories, also in the pypi pyproject the repository is linked.

Regarding the deprecancy of the 3DS you mentioned, I made a mesh export with `.glb` files and they seem to have no smoothing groups, at least no smoothing option, wich other `.mvr` importes could use. I tried it with MA3 on PC and `.glb` do not render as nicely as `.3ds` files. The only advantage would be the image packing, but that also is not supported by every importer and you can pack the images directly into the `.mvr` zip. Thats why I am sticking to `.3ds`

I also made a little change into the mvr.py. If a model has parents, only the parent has to be transformed.  e55ab8de343e6fffcab5c1b78134d2249e6cdce3

nrgsille76 commented:
BTW It's now ready again for review ;)

vanous commented:
Thank you.

> The only advantage would be the image packing, but that also is not supported by every importer

Which other importers have you tried? Strange that they would not support images from glb, but would support them from 3ds...

As for the focus points - is this for objects that have a focus point, but also have child geometries? Do you have a testing mvr for that?

Also, thank you for re-formatting, this is helpful to keep attribution correct.

I need to address the elephant in the room again: why having two Blender import/export GDTF&MVR extensions, when most of the code is the same anyways? You could still do your FB posts and so on... You could bring your ideas into Blender DMX, it will eliminate duplication, and bugs that currently exist...

vanous commented:
> > As i mentioned, it is being [required from us by the Extension store maintainers](https://extensions.blender.org/approval-queue/open-stage-blender-dmx/#activity-1547). Please use the same version as is officially present on Pypi.
> 
> That's strange, I only created it one time and then uploaded it and there is no difference in the code. Anyways, I have now downloaded directly from pypi and pushed it in the PR.

I added a CI check for this now. Please rebase onto main again. Do not merge main into your branch, that messes up the commit graph, simply rebase your branch onto main.

> > Which other importers have you tried? Strange that they would not support images from glb, but would support them from 3ds...
> > As for the focus points - is this for objects that have a focus point, but also have child geometries? Do you have a testing mvr for that?
> I have tried the MA3 importer and the importer from Depence. Images getting loaded if they are inside the mvr zip but not if they are packed into the glb file. 

Interesting. Have you reported this to Depence? Would make a sense to report, if you can reproduce this.

> For the focus points, [they can include a Geometries node](https://www.gdtf.eu/mvr/mvr-spec/file-format-definition/#node-definition-focuspoint). One of the files you can use for testing is the "scene_objects.mvr" wich I got from you. Usually the meshes should import at focus point position, this has not worked accuratly in the tests with mvr.py. Do you have any function that reparents the objects? I have not found out why its working on my code and not with yours.

We do re-parent only very direct fixture to a direct parent connections (not via group objects, but via child_list), so if a truss has a child_list of fixtures, the fixtures become children of the truss, so when you move the truss, the fixtures follow. But even if you comment out lines 729-731 in mvr.py, the objects linked to the focus points in the scene_objects.mvr do not appear.

> I can explain the elephant in the room: BlenderDMX has more focus on the fixtures and the networking and controlling. My extension has another concept, the focus is on the meshes and environment, to put meshes and fixtures in position and then export that easily as .mvr, I also will further provide my ideas for this project, but I do not have the complete overview over the code, thats why the workflow is more efficient for me when developing on a seperated project.

OK. I was trying to do this as a welcoming gesture, to eliminate user confusion, code copying, maintenance and bugs.
